functions.php

```diff
@@ -719,10 +732,54 @@ function mozilla_update_member() {
                 unset($required[0]);
 
                 foreach($required AS $field) {
+
                     $form_data = sanitize_text_field(trim($_POST[$field]));
                     update_user_meta($user->ID, $field, $form_data);
                 }
             }
         }
     }
 }
+
+function mozilla_is_logged_in() {
+    $current_user = wp_get_current_user()->data;
+    return sizeof((Array)$current_user) > 0 ? true : false; 
+}
+
+
+function mozilla_get_user_visibility_settings($user_id) {
+    $user = get_user_by('ID', $user_id);
+    $meta = get_user_meta($user_id);
+
+    $visibility_fields = Array(
+                                'username',
+                                'first_name',
+                                'last_name',
+                                'email'
+    );
+
+    $visibility_settings = Array();
+
+    foreach($visibility_fields AS $field) {
+        if(isset($meta["{$field}_visibility"][0])) {
+            $visibility_settings["{$field}_visibility"] = intval($meta["{$field}_visibility"][0]);
+        } else {
+            if($field === 'username') {
+                $visibility_settings["{$field}_visobility"] = PrivacySettings::PUBLIC_USERS;
+            } else {
+                $visibility_settings["{$field}_visibility"] = PrivacySettings::REGISTERED_USERS;
+            }
+        }
+    }
+
+    return $visibility_settings;
+}
+
+function mozilla_save_event($post_id, $post, $update) {
+  if ($post->post_type === 'event') {
+    $event = new stdClass();
+    $event->image_url = $_POST['image_url'];
```

Comment:
I should have mentioned this when we were going over this we should always sanitize the data that user's are giving us through a form, we never know if the user has some kind of malicious intent.  We should try to avoid putting data into the database straight from $_POST / $_GET.  Wordpress has functions we can leverage such as
https://developer.wordpress.org/reference/functions/sanitize_text_field/.  Here is a list of all functions wordpress has for validating and sanitizing data https://codex.wordpress.org/Validating_Sanitizing_and_Escaping_User_Data.